packages/web3-common/src/eth_execution_api.ts

```diff
@@ -0,0 +1,293 @@
+	readonly uncles: Uncles;
+	readonly hash: HexString32Bytes | null;
+}
+
+// https://github.com/ethereum/execution-apis/blob/main/src/schemas/filter.json#L59
+export type Topic = HexString256Bytes;
+
+// https://github.com/ethereum/execution-apis/blob/main/src/schemas/receipt.json#L2
+export interface Log {
+	readonly removed?: boolean;
+	readonly logIndex?: Uint;
+	readonly transactionIndex?: Uint;
+	readonly transactionHash?: HexString32Bytes;
+	readonly blockHash?: HexString32Bytes;
+	readonly blockNumber?: Uint;
+	readonly address?: Address;
+	readonly data?: HexStringBytes;
+	readonly topics?: Topic[];
+}
+
+// https://github.com/ethereum/execution-apis/blob/main/src/schemas/receipt.json#L44
+export interface ReceiptInfo {
+	readonly transactionHash: HexString32Bytes;
+	readonly transactionIndex: HexString32Bytes;
+	readonly blockHash?: HexString32Bytes;
+	readonly blockNumber?: Uint;
+	readonly from?: Address;
+	readonly to?: Address;
+	readonly cumulativeGasUsed?: Uint;
+	readonly gasUsed?: Uint;
+	readonly contractAddress?: Address;
+	readonly logs?: Log[];
+	readonly logsBloom?: HexString256Bytes;
+}
+
+// https://github.com/ethereum/execution-apis/blob/main/src/schemas/client.json#L2
+export type SyncingStatus =
+	| { startingBlock: Uint; currentBlock: Uint; highestBlock: Uint }
+	| boolean;
+
+// https://github.com/ethereum/execution-apis/blob/main/src/eth/fee_market.json#L53
+export interface FeeHistoryResult {
+	readonly oldestBlock: Uint;
+	readonly baseFeePerGas: Uint;
+	readonly reward: number[][];
+}
+
+// https://github.com/ethereum/execution-apis/blob/main/src/schemas/filter.json#L28
+export interface Filter {
+	readonly fromBlock?: Uint;
+	readonly toBlock?: Uint;
+	readonly address?: Address | Address[];
+	readonly topics?: Topic[];
+}
+
+// https://github.com/ethereum/execution-apis/blob/main/src/schemas/filter.json#L2
+export type FilterResults = HexString32Bytes[] | Log[];
+
+export interface CompileResult {
+	readonly code: HexStringBytes;
+	readonly info: {
+		readonly source: string;
+		readonly language: string;
+		readonly languageVersion: string;
+		readonly compilerVersion: string;
+		readonly abiDefinition: Record<string, unknown>[];
+		readonly userDoc: {
+			readonly methods: Record<string, unknown>;
+		};
+		readonly developerDoc: {
+			readonly methods: Record<string, unknown>;
+		};
+	};
+}
+
+/* eslint-disable camelcase */
+export type EthExecutionAPI = {
+	// https://github.com/ethereum/execution-apis/blob/main/src/eth/block.json
+	eth_getBlockByHash: (blockHash: HexString32Bytes, hydrated: boolean) => Block;
+	eth_getBlockByNumber: (blockNumber: BlockNumberOrTag, hydrated: boolean) => Block;
+	eth_getBlockTransactionCountByHash: (blockHash: HexString32Bytes) => Uint;
+	eth_getBlockTransactionCountByNumber: (blockNumber: BlockNumberOrTag) => Uint;
+	eth_getUncleCountByBlockHash: (blockHash: HexString32Bytes) => Uint;
+	eth_getUncleCountByBlockNumber: (blockNumber: Uint) => Uint;
```

Comment:
```suggestion
	eth_getUncleCountByBlockNumber: (blockNumber: BlockNumberOrTag) => Uint;
```

Opened an [issue](https://github.com/ethereum/execution-apis/issues/114) for this